test_util/src/lsp.rs

```diff
@@ -78,7 +78,7 @@ where
         .ok_or_else(|| anyhow::anyhow!("missing capture"))?;
       content_length = content_length_match.as_str().parse::<usize>()?;
     }
-    if &buf == "\r\n" {
+    if &buf == "\r\n" || buf.is_empty() {
```

Comment:
I think we should panic in this case. I don't think we want to fail silently for invalid data in a test or is there some reason to?

Reply:
Considering review,
>the language server does not produce messages without a content-length: lead in. If it ever did, we would have bigger issues and an infinite loop would be an acceptable failure state for the test.

I fixed it to just add invalid case.